pyiceberg/manifest.py

```diff
@@ -389,6 +389,10 @@ def __eq__(self, other: Any) -> bool:
         """
         return self.file_path == other.file_path if isinstance(other, DataFile) else False
 
+    def update(self, other: Dict[str, Any]) -> None:
```

Comment:
I like to avoid mutability. Looking at the code, I think we can also create the object at once:

```python
data_file = DataFile(
        content=DataFileContent.DATA,
        file_path=file_path,
        file_format=FileFormat.PARQUET,
        partition=Record(),
        file_size_in_bytes=len(fo),
        # After this has been fixed:
        # https://github.com/apache/iceberg-python/issues/271
        # sort_order_id=task.sort_order_id,
        sort_order_id=None,
        # Just copy these from the table for now
        spec_id=table_metadata.default_spec_id,
        equality_ids=None,
        key_metadata=None,
        **statistics.to_serialized_dict()
    )
```